cmd/clusterctl/client/cluster/ownergraph.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cluster
+
+import (
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// OwnerGraph contains a graph with all the objects considered by clusterctl move as nodes and the OwnerReference relationship
+// between those objects as edges.
+type OwnerGraph map[string]OwnerGraphNode
+
+// OwnerGraphNode is a single node linking an ObjectReference to its OwnerReferences.
+type OwnerGraphNode struct {
+	Object corev1.ObjectReference
+	Owners []metav1.OwnerReference
+}
+
+func nodeToOwnerRef(n *node, attributes ownerReferenceAttributes) metav1.OwnerReference {
+	ref := metav1.OwnerReference{
+		Name:       n.identity.Name,
+		APIVersion: n.identity.APIVersion,
+		Kind:       n.identity.Kind,
+		UID:        n.identity.UID,
+	}
+	if attributes.BlockOwnerDeletion != nil {
+		ref.BlockOwnerDeletion = attributes.BlockOwnerDeletion
+	}
+	if attributes.Controller != nil {
+		ref.Controller = attributes.Controller
+	}
+	return ref
+}
+
+// GetOwnerGraph returns a graph with all the objects considered by clusterctl move as nodes and the OwnerReference relationship between those objects as edges.
+// NOTE: this data structure is exposed to allow implementation of E2E tests verifying that CAPI can properly rebuild its
+// own owner references; there is no guarantee about the stability of this API.
+func GetOwnerGraph(namespace, kubeconfigPath string) (OwnerGraph, error) {
```

Comment:
```suggestion
func GetOwnerGraph(namespace, kubeconfigPath string) (*OwnerGraph, error) {
```
nit: up to you - maybe easier to handle (and I guess a lot less copying around of data)

Reply:
I don't think it would make it easier to handle really, and given this is a test func I think it's alright as is. No strong feelings though if you want to change it.

Reply:
No, up to you, all good :0